backend/kernelCI_app/utils.py

```diff
@@ -152,3 +152,19 @@ def validate_comparison_op(self, op):
     def get_comparison_op(self, filter, op_type="orm"):
         idx = self.comparison_op_type_idx[op_type]
         return self.comparison_ops[filter["comparison_op"]][idx]
+
+    def get_grouped_filters(self):
+        grouped_filters = {}
+
+        for f in self.filters:
+            field = f["field"]
+            value = f['value']
+            if field not in grouped_filters:
+                grouped_filters[field] = f
+                grouped_filters[field]['value'] = value
+            elif type(grouped_filters[field]['value']) is str:
+                grouped_filters[field]['value'] = [grouped_filters[field]['value'], value]
```

Comment:
from my understanding, this is checking if it is a string and if it is a string it converts to an array

shouldn't it always be an array? so you don't have the string check 

```suggestion
                grouped_filters[field]['value'] = [value]
```